### ISSUE TITLE
clarification question about design files

### ISSUE THREAD

**Issue by iancballard** [mwaskom/lyman#65]
When you include values for regressor in the design file, does lyman create two regressors (the linear and constant term)? If so does it take care of orthogonalization? Or should I specify this myself?

**Comment by mwaskom** [comment]
The latter.

**Comment by kastman** [comment]
@iancballard Did you have success orthogonalizing the values for input into the design file, or figure out another way to orthogonalize besides passing a fully formed condition matrix into the additional regressors? I get how to orthogonalize a full matrix after its been constructed, but I'm not sure about back-projecting that to adjust the values themselves for the design. Any good suggestions? Thanks!

**Comment by iancballard** [comment]
Hello. For parametric analyses it's very straightforward: simply make sure your parametric values are mean-centered, and they will be orthogonal to a unit vector. For PPI analysis it's a bit more complex, because you are entering a regressor that has already been convolved. For these cases, I just wrote up a simple vector rejection (https://en.wikipedia.org/wiki/Vector_projection) script to orthogonalize the interaction regressor before entering it.

**Comment by kastman** [comment]
That's so obvious it's embarrassing to have asked; thanks for your quick reply.

For the PPI, that would be a rejection of the task vector in matrix form on the interaction regressor, right? I poked in your public repos but didn't see an example script; would you have a gist that you could point me to? I realized that I've been including PPI regressors without orthogonalizing them.

**Comment by iancballard** [comment]
Ah I spoke without thinking. Rather than orthogonalizing, I centred following the guidlines in http://fsl.fmrib.ox.ac.uk/fsl/fslwiki/PPIZeroing. I did orthogonalize the physiological regressor against my white matter and ventricle noise regressors, hoping that it would speed up computation if the design matrix was less rank deficient. Though i'm not sure if that is actually the case. Here's an example PPI script that I used with lyman: 

https://github.com/iancballard/fd_fmri/blob/master/ppi_analysis.ipynb

It contains a vector_rejection function which can be used for orthogonalizing in general.
